File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        a = 0
        hold = set()

        lngth = 0
        for i in range(len(s)):
            while s[i] in hold:
                hold.remove(s[a])
                a += 1
            hold.add(s[i])
            lngth = max(len(hold), lngth)
        return lngth

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let hold = new Set();
    let a = 0;
    let lnth = 0;
    for(let i = 0; i < s.length; i++){
        while(hold.has(s[i])){
            hold.delete(s[a]);
            a++;
        }
        hold.add(s[i]);
        lnth = Math.max(lnth, i-a+1);
    }
    return lnth;
};